### PR TITLE
fix 4078 / race condition 

### PR DESCRIPTION
**Pull request by meejah**
Fix:ticket:#4078

### CODE REVIEW

File: src/allmydata/test/cli/test_run.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def test_non_numeric_pid(self):
         self.assertThat(runs, Equals([]))
         self.assertThat(result_code, Equals(1))
 
-    good_file_content_re = re.compile(r"\s[0-9]*\s[0-9]*\s", re.M)
+    good_file_content_re = re.compile(r"\s*[0-9]*\s[0-9]*\s*", re.M)
```
Comment (itamarst) [inline src/allmydata/test/cli/test_run.py:267]: ... is this relevant to the fixes?

Comment (meejah) [reply src/allmydata/test/cli/test_run.py:267]: Not really at all no, but Hypothesis found a failing case `'0\r0'` which the previous regex didn't match. Could put in a different bug+PR if this one ... keeps failing?

**Comment by meejah** [comment]
@itamarst in IRC you said that the `Reactor was unclean` errors are known? Do they consistently cause `test_upload_and_download_convergent` to fail (as with the Ubuntu builder currently) and thus I should ignore them for this particular "fix"? Or is the output just confusing in CircleCI?

**Comment by meejah** [comment]
The integration failure seems to be a single Tor node failing to start. Haven't really repeated locally (except that "sometimes" my local Chutney fails to bootstrap within the 600 seconds).

File: src/allmydata/test/cli/test_run.py (hunk 1 above)
Comment (meejah) [reply src/allmydata/test/cli/test_run.py:267]: Want me to split this out..?

**Review by itamarst — APPROVED** [review]
